test_plans/weak_object.asciidoc

```diff
@@ -0,0 +1,212 @@
+  ** Verify that `SyclObject.ext_oneapi_owner_before(w1) == false` and
+  ** Verify that `SyclObject.ext_oneapi_owner_before(w2) == false`.
+
+* Verify that `owner_before` compares equivalent for two weak objects that are both empty:
+  ** Create a SYCL object of type `SyclObjT`. Call it `SyclObject`.
+  ** Create two empty weak_object objects by using the default constructor. Call them `w1` and `w2`.
+  ** Verify that `w1.owner_before(w2) == false` and `w2.owner_before(w1) == false`.
+  ** Verify that `SyclObject.ext_oneapi_owner_before(w1) == true` or
+  ** Verify that `SyclObject.ext_oneapi_owner_before(w2) == true`.
+
+* Verify that `owner_before` has some order for two weak object that refer to different underlying SYCL objects:
+  ** Create two SYCL objects of type `SyclObjT`. Call them `SyclObject1` and `SyclObject2`.
+  ** Create a `weak_object` object from `SyclObject1` and another from `SyclObject2`. Call them `w1` and `w2`.
+  ** Verify that exactly one of the following is `true` for weak objects:
+    *** `w1.owner_before(w2) == true && w2.owner_before(w1) == false`, or
+    *** `w1.owner_before(w2) == false && w2.owner_before(w1) == true`.
+  ** Verify that exactly one of the following is `true` for SYCL objects:
+    *** `SyclObject1.ext_aoneapi_owner_before(w2) == true && SyclObject2.ext_aoneapi_owner_before(w1) == false`, or
+    *** `SyclObject1.ext_aoneapi_owner_before(w2) == false && SyclObject2.ext_aoneapi_owner_before(w1) == true`.
+  ** Verify that `w1.owner_before(SyclObject2) == w1.owner_before(w2)`
+  ** Verify that `w2.owner_before(SyclObject1) == w2.owner_before(w1)`
+  ** Verify that `SyclObject1.ext_oneapi_owner_before(SyclObject2) == Sycl_object1.ext_oneapi_owner_before(w2)`
+  ** Verify that `SyclObject2.ext_oneapi_owner_before(SyclObject1) == Sycl_object2.ext_oneapi_owner_before(w1)`
+
+=== owner_less
+
+Similar to previous section.
+
+* Verify that `owner_less` compares equivalent for two weak objects that both refer to the same underlying SYCL object:
+  ** Create a SYCL object of type `SyclObjT`. Call it `SyclObject`.
+  ** Create two weak_object objects from SyclObject. Call them `w1` and `w2`.
+  ** Verify that:
+    *** `ext::oneapi::owner_less(w1, w2) == false`.
+    *** `ext::oneapi::owner_less(w2, w1) == false`.
+    *** `ext::oneapi::owner_less(w1, SyclObject) == false`.
+    *** `ext::oneapi::owner_less(SyclObject, w1) == false`.
+    *** `ext::oneapi::owner_less(w2, SyclObject) == false`.
+    *** `ext::oneapi::owner_less(SyclObject, w2) == false`.
+
+* Verify that `owner_less` compares equivalent for two weak objects that are both empty:
+  ** Create a SYCL object of type `SyclObjT`. Call it `SyclObject`.
+  ** Create two empty weak_object objects by using the default constructor. Call them `w1` and `w2`.
+  ** Verify that `ext::oneapi::owner_less(w1, w2) == false`.
+  ** Verify that `ext::oneapi::owner_less(w1, SyclObject) == true` or `ext::oneapi::owner_less(w2, SyclObject) == true`
```

Comment:
This test is not correct.  Since "SyclObject" is never empty, you cannot test that it is equivalent to an empty weak pointer.  I'd just remove this line.